crates/metrics/metrics-derive/src/expand.rs

```diff
@@ -93,37 +129,56 @@ pub(crate) fn derive(node: &DeriveInput) -> Result<proc_macro2::TokenStream> {
-                        format!("{}{}{}", scope, #separator, #name)
-                    };
-                    let field_name = &metric.field.ident;
-
-                    let registrar = metric.register_stmt()?;
-                    let describe = metric.describe_stmt()?;
-                    let description = &metric.description;
-
-                    Ok((
-                        quote! {
-                            #field_name: #registrar(#metric_name),
-                        },
-                        quote! {
-                            #field_name: #registrar(#metric_name, labels.clone()),
-                        },
-                        quote! {
-                            #describe(#metric_name, #description);
-                        },
-                    ))
+                    let field_name = &metric.field().ident;
+                    match metric {
+                        MetricField::Included(metric) => {
+                            let name = metric.name();
+                            let separator = metrics_attr.separator();
+                            let metric_name = quote! {
+                                format!("{}{}{}", scope, #separator, #name)
+                            };
+
+                            let registrar = metric.register_stmt()?;
+                            let describe = metric.describe_stmt()?;
+                            let description = &metric.description;
+
+                            Ok((
+                                quote! {
+                                    #field_name: #registrar(#metric_name),
+                                },
+                                quote! {
+                                    #field_name: #registrar(#metric_name, labels.clone()),
+                                },
+                                Some(quote! {
+                                    #describe(#metric_name, #description);
+                                }),
+                            ))
+                        }
+                        MetricField::Skipped(_) => Ok((
+                            quote! {
+                                #field_name: Default::default(),
+                            },
+                            quote! {
+                                #field_name: Default::default(),
+                            },
+                            None,
+                        )),
+                    }
                 })
                 .collect::<Result<Vec<_>>>()?
                 .into_iter()
-                .fold((vec![], vec![], vec![]), |mut acc, x| {
-                    acc.0.push(x.0);
-                    acc.1.push(x.1);
-                    acc.2.push(x.2);
-                    acc
-                });
+                .fold(
+                    (vec![], vec![], vec![]),
+                    |(mut defaults, mut labeled_defaults, mut describes),
+                     (default, labeled_default, describe)| {
```

Comment:
let's avoid using keywords as var names

Reply:
ah, didn't know it's a keyword from nightly feature, fixed